webcurator-core/src/main/java/org/webcurator/core/util/ProcessBuilderUtils.java

```diff
@@ -28,9 +32,63 @@ public static String getFullPathOfCommand(String commandName) {
             log.error("Unable to process the command in a new thread.", e);
         }
 
-        if (ret.size() > 0 && ret.get(0).endsWith(commandName)) {
+        if (!ret.isEmpty() && ret.get(0).endsWith(commandName)) {
             return ret.get(0);
         }
         return null;
     }
-}
+
+    public static int processCommand(File targetDirectory, String[] commands) {
+        List<String> commandList = Arrays.asList(commands);
+        ProcessBuilder processBuilder = new ProcessBuilder(commandList);
+        processBuilder.directory(targetDirectory);
+        try {
+            Process process = processBuilder.inheritIO().start();
+            int processStatus = process.waitFor();
+
+            if (processStatus != 0) {
+                log.error("Process ended with a failed status: {}, commands: {}", processStatus, String.join(",", commands));
+            }
+            return processStatus;
+        } catch (Exception e) {
+            log.error("Unable to process the command in a new thread.", e);
+            return UNKNOWN_ERROR;
+        }
+    }
+
+    public static int wbManagerInitCollection(File targetDirectory, String collName) {
+        String wb_manager = ProcessBuilderUtils.getFullPathOfCommand("wb-manager");
+        if (wb_manager == null) {
+            log.error("The command tool [wb-manager] is not installed or not in the env PATH.");
+            return UNKNOWN_ERROR;
+        }
+        String[] commands = {wb_manager, "init", collName};
+        return processCommand(targetDirectory, commands);
+    }
+
+    public static int wbManagerAddWarcFile(File targetDirectory, String collName, String warcFilePath) {
+        String wb_manager = ProcessBuilderUtils.getFullPathOfCommand("wb-manager");
+        if (wb_manager == null) {
+            log.error("The command tool [wb-manager] is not installed or not in the env PATH.");
+            return UNKNOWN_ERROR;
+        }
+        String[] commands = {wb_manager, "add", collName, warcFilePath};
+        return processCommand(targetDirectory, commands);
+    }
+
+    public static boolean forceDeleteDirectory(File toBeDeletedDirectory) {
+        if (toBeDeletedDirectory == null || !toBeDeletedDirectory.exists()) {
+            return true;
+        }
+        String os = SystemUtils.OS_NAME.toLowerCase();
+        if (os.contains("linux") || os.contains("unix")) {
```

Comment:
To be (fairly) complete, it would probably be better to use SystemUtils.IS_OS_UNIX || os.contains("bsd") (the commons-lang people seem to think Max OS X is the only BSD that qualifies as a Unix family member).